packages/flutter/lib/src/material/popup_menu.dart

```diff
@@ -41,7 +41,11 @@ class PopupMenuDivider extends PopupMenuEntry<dynamic> {
 
   final double height;
 
-  Widget build(BuildContext context) => new Divider(height: height);
+  _PopupMenuDividerState createState() => new _PopupMenuDividerState();
+}
+
+class _PopupMenuDividerState extends State<PopupMenuDivider> {
+  Widget build(BuildContext context) => new Divider(height: config.height);
```

Comment:
You can probably get away with having an empty PopupMenuEntry state that everyone uses except `CheckedPopupMenuItem`. Something like:

```dart
class PopupMenuEntryState<T, U extends PopupMenuEntry<T>> extends State<U> {
  Widget build(BuildContext context) => config.build(context);
}
```

Then in PopupMenuEntry:

```dart
  PopupMenuEntryState<T, PopupMenuEntry> createState() => new PopupMenuEntryState<T, PopupMenuEntry>();
```

Then all the build functions can stay on the components, no need for any states except CheckedPopupMenuItem, which would do:

```dart
  _CheckedPopupMenuItemState<T> createState() => new _CheckedPopupMenuItemState<T>();
```

...and _CheckedPopupMenuItemState would be defined as:

```dart
class _CheckedPopupMenuItemState<T> extends PopupMenuEntryState<T, CheckedPopupMenuItem<T>> { ... }
```